### PR TITLE
Add Linux Build and CI Support

### PR DESCRIPTION
**Pull request by brianrob**
1. Add a new script to build the CLI based components on Linux.
2. Update the CI jobs to build on Ubuntu in addition to Windows.

NOTE: This change was tested in the citest branch - it won't take effect on the CI jobs for this PR.

### CODE REVIEW

**Comment by brianrob** [comment]
@mmitche, PTAL

@lt72, FYI

**Comment by brianrob** [comment]
@dotnet-bot test Debug Build please

**Comment by brianrob** [comment]
@dotnet-bot test Debug Build please

**Comment by mmitche** [comment]
@brianrob what were the other two issues you saw there?

File: netci.groovy [hunk 1]
```diff
@@ -4,26 +4,77 @@ import jobs.generation.Utilities;
...
+            def winOS = 'Windows_NT';
+            def winBuildJobName = "${winOS.toLowerCase()}_${lowerConfigurationName}";
+            def newWinJob = job(Utilities.getFullJobName(project, winBuildJobName, isPR)) {
+                steps {
+                    batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\Tools\\VsDevCmd.bat\" && CIBuild /${lowerConfigurationName}")
+                    batchFile("C:\\Packer\\Packer.exe .\\LocalPackages.pack . LocalPackages")
+                }
+            }
+
+            Utilities.setMachineAffinity(newWinJob, "${winOS}", 'latest-or-auto')
```
Comment (mmitche) [inline netci.groovy]: you don't need the quotes around winOS here

File: netci.groovy [hunk 2]
```diff
@@ -4,26 +4,77 @@ import jobs.generation.Utilities;
...
+                    shell("chmod +x BuildCliComponents.sh")
+                    shell("./BuildCliComponents.sh ${lowerConfigurationName}")
+                }
+
+                parameters {
+                    stringParam('WIN_BUILD', '', 'Build number to use for copying binaries for Linux build.')
+                }
+            }
+
+            Utilities.setMachineAffinity(newLinuxJob, "${os}", 'latest-or-auto')
```
Comment (mmitche) [inline netci.groovy]: You don't need the quotes around os here

File: netci.groovy [hunk 3]
```diff
@@ -4,26 +4,77 @@ import jobs.generation.Utilities;
...
+            def fullXunitPerformanceTestJobName = projectFolder + '/' + newLinuxJob.name
+            def newLinuxFlowJob = buildFlowJob(Utilities.getFullJobName(project, linuxFlowJobName, isPR)) {
+                buildFlow("""
+                    b = build(params, '${fullWinBuildJobName}')
+                    build(params +
+                    [WIN_BUILD: b.build.number], '${fullXunitPerformanceTestJobName}')
+                    """)
+            }
+
+            Utilities.setMachineAffinity(newLinuxFlowJob, "${os}", 'latest-or-auto')
```
Comment (mmitche) [inline netci.groovy]: You don't need the quotes around os here

File: netci.groovy [hunk 4]
```diff
@@ -4,26 +4,77 @@ import jobs.generation.Utilities;
...
+            Utilities.setMachineAffinity(newLinuxJob, "${os}", 'latest-or-auto')
+            Utilities.standardJobSetup(newLinuxJob, project, isPR, "*/${branch}")
+
+            // Setup a flow job to orchestrate things.
+            def fullXunitPerformanceTestJobName = projectFolder + '/' + newLinuxJob.name
+            def newLinuxFlowJob = buildFlowJob(Utilities.getFullJobName(project, linuxFlowJobName, isPR)) {
+                buildFlow("""
+                    b = build(params, '${fullWinBuildJobName}')
+                    build(params +
+                    [WIN_BUILD: b.build.number], '${fullXunitPerformanceTestJobName}')
```
Comment (mmitche) [inline netci.groovy]: The quotes are wrong here (no inline replacements in the singly quoted strings

**Comment by mmitche** [comment]
LGTM aside from the minor nits

**Comment by brianrob** [comment]
@dotnet-bot test ci please

**Comment by brianrob** [comment]
@mmitche, thanks.  I've fixed all of these.

**Comment by brianrob** [comment]
@dotnet-bot test Release Build

**Comment by brianrob** [comment]
@dotnet-bot test ci please

**Comment by brianrob** [comment]
@dotnet-bot test Debug Build

**Comment by brianrob** [comment]
@dotnet-bot test Debug Build

**Pull request merged**